python/pyarrow/lib.pxd

```diff
@@ -65,7 +65,9 @@ cdef class StructType(DataType):
     cdef:
         const CStructType* struct_type
 
-    cdef Field child_by_name(self, name)
+    cpdef Field child_by_name(self, name)
+    cpdef Field field(self, int i)
+    cpdef Field field_by_name(self, name)
```

Comment:
I'm not sure it's useful to expose these to Python ("cpdef").

Reply:
I guess the status quo is using `__getitem__` to access the children. I'll revert this change for now (easier to add APIs later than take them away)